TypeScript-Playground/tests/typescript-assignment.test.ts

```diff
@@ -0,0 +1,124 @@
+import { test, expect, describe } from "vitest";
+import {
+  generateFibocacciSeries,
+  map,
+  filter,
+  reduce,
+  vitestExpect,
+  processQuery,
+  LinkedList,
+  address,
+} from "../src/typescript-assignment";
+
+/**
+ * Test suite for generic utility functions map, filter, and reduce.
+ */
+describe("Tests for assignment questions on typescript", () => {
+  /**
+   * Test case for generating fibonacci Series
+   */
+  test("Test for generating fibonacci series", () => {
+    expect(generateFibocacciSeries(10)).toEqual([
+      0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55,
+    ]);
+  });
+  /**
+   * Test case for the map function.
+   * Converts numbers to strings.
+   */
+  test("Tests for generics using map function", () => {
+    /**
+     * Function to convert a number to a string.
+     * @param num - The number to convert.
+     * @returns The number as a string.
+     */
+    function converToString(num: number): string {
+      return num.toString();
+    }
+    expect(map([1, 2, 3], converToString)).toEqual(["1", "2", "3"]);
+  });
+  /**
+   * Test case for the filter function.
+   * Filters out odd numbers from an array of numbers.
+   */
+  test("Tests for generics using filter function", () => {
+    /**
+     * Function to check if a number is even.
+     * @param num - The number to check.
+     * @returns True if the number is even, false otherwise.
+     */
+    function isEven(num: number): boolean {
+      return num % 2 === 0;
+    }
+    expect(filter([2, 3, 4, 5, 6], isEven)).toEqual([2, 4, 6]);
+  });
+  /**
+   * Test case for the reduce function.
+   * Sums an array of numbers.
+   */
+  test("Tests for generics using reduce function", () => {
+    /**
+     * Function to sum two numbers.
+     * @param acc - The accumulator.
+     * @param value - The current value.
+     * @returns The sum of the accumulator and the current value.
+     */
+    function sum(acc: number, value: number): number {
+      return acc + value;
+    }
+    expect(reduce([2, 4, 6], sum, 0)).toEqual(12);
+  });
+  /**
+   * Test case for the vitestExpect function.
+   * Tests assertions using vitestExpect.
+   */
+  test("Tests for implementing expect function of vitest", () => {
+    expect(() => vitestExpect(5).toBe(5)).not.toThrow();

```

Comment:
no need of expect, we could directly use vitestExpect (though this naming is misleading). may be if you want to differentiate, use customExpect

Reply:
Implemented the changes.